filestore/file.go

```diff
@@ -1,28 +1,35 @@
 package filestore
 
-import "os"
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+)
 
 type fd struct {
 	*os.File
 	filename string
+	basepath string
 }
 
-func newFile(filename Path) (File, error) {
+func newFile(basepath, filename Path) (File, error) {
 	var err error
-	result := fd{filename: string(filename)}
-	result.File, err = os.OpenFile(result.filename, os.O_APPEND|os.O_CREATE|os.O_RDWR, 0644)
+	result := fd{filename: string(filename), basepath: string(basepath)}
+	full := path.Join(string(basepath), string(filename))
+	result.File, err = os.OpenFile(full, os.O_APPEND|os.O_CREATE|os.O_RDWR, 0644)
 	if err != nil {
 		return nil, err
 	}
 	return &result, nil
 }
 
 func (f fd) Path() Path {
-	return Path(f.filename)
+	return Path(f.Name())
 }
 
 func (f fd) Size() int64 {
-	info, err := os.Stat(f.filename)
+	info, err := os.Stat(fmt.Sprintf("%s%c%s", f.basepath, filepath.Separator, f.filename))
```

Comment:
Seems like we are creating a full path a few times a few different ways; we could do...

```suggestion
	info, err := os.Stat(f.absPath())
```
if we had a function:

```go
func (f fd) absPath() string {
	return path.Join(f.basepath, f.filename)
}
```